backend/src/rtp.rs

```diff
@@ -568,32 +583,30 @@ impl<T: BorrowMut<[u8]>> Packet<T> {
         &mut self.serialized_mut()[payload_range]
     }
 
-    // TODO: Return a Result instead
     // pub for tests
-    pub fn decrypt_in_place(&mut self, key: &Key, salt: &Salt) -> Option<()> {
+    pub fn decrypt_in_place(&mut self, key: &Key, salt: &Salt) -> Result<(), PacketError> {
```

Comment:
I am seeing a warning when running tests. For clippy, did you run this? It should catch them all:
`cargo clippy --all-features --all-targets -- -D warnings`